starport/pkg/placeholder/tracer.go

```diff
@@ -0,0 +1,129 @@
+package placeholder
+
+import (
+	"strings"
+
+	"github.com/tendermint/starport/starport/pkg/validation"
+)
+
+type iterableStringSet map[string]struct{}
+
+func (set iterableStringSet) Iterate(f func(i int, element string) bool) {
+	i := 0
+	for key := range set {
+		if !f(i, key) {
+			return
+		}
+		i++
+	}
+}
+
+func (set iterableStringSet) Add(item string) {
+	set[item] = struct{}{}
+}
+
+var _ validation.Error = (*ErrMissingPlaceholders)(nil)
+
+type ErrMissingPlaceholders struct {
```

Comment:
```suggestion
type MissingPlaceholdersError struct {
```

by the Go conventions.